app/common/utilities.py

```diff
@@ -122,5 +119,23 @@ def add_to_and_from_details(messages):
     for message in messages:
         message['msg_to'][0] = next((user for user in user_details if user["id"] == message['msg_to'][0]), None)
         message['msg_from'] = next((user for user in user_details if user["id"] == message['msg_from']), None)
+
+    messages = add_business_details(messages)
+    return messages
+
+
+def add_business_details(messages):
+    """Adds business details"""
+
+    ru_list = []
+
+    for message in messages:
+        if message['ru_ref'] not in ru_list:
+            ru_list.append(message['ru_ref'])
+
+    business_details = business_by_ru.get_business_details_by_ru(ru_list)
+
+    for message in messages:
+        message['ru_ref'] = next((user for user in business_details if user["ru_ref"] == message['ru_ref']), None)
```

Comment:
'user' needs to be renamed to something else, maybe 'bussiness'. Also might be better if instead of 'ru_ref' being a dictionary with 'ru_ref' and 'business_name', have 'ru_ref' and 'business_name' as separate keys.